mcv/patchwork.py

```diff
@@ -0,0 +1,150 @@
+"""Patchwork"""
+import os
+import os.path
+import json
+import requests
+import subprocess
+
+from functools import partial
+
+PATCHWORK_API_ENDPOINT = os.getenv('PATCHWORK_API_ENDPOINT',
+                                   "https://api.patchworksecurity.com/api/v1")
+DEFAULT_UUID_PATH = '/etc/patchwork/uuid'
+DPKG_QUERY_CMD = ["/usr/bin/dpkg-query", "-W",
+                  "-f", "${Status}\t${Package}\t${Version}\n"]
+
+
+def register(api_key, name, uuid_path=DEFAULT_UUID_PATH):
+    """Register this machine
+
+    UUID metadata is stored in uuid_path. The machine is only
+    registered if the UUID metadata isn't found.
+
+    Returns:
+        dict: JSON metadata for this machine
+    """
+    if os.path.exists(uuid_path):
+        with open(uuid_path, 'r') as f:
+            machine_data = json.load(f)
+
+        return machine_data
+
+    _mkdirs(uuid_path)
```

Comment:
Would expect `_mkdirs` to be defined above this function if this function depends upon it